lib/driver.js

```diff
@@ -734,31 +734,29 @@ class XCUITestDriver extends BaseDriver {
       return {device, realDevice: true, udid: this.opts.udid};
     }
 
-    // figure out the correct simulator to use, given the desired capabilities
-    let device = await getExistingSim(this.opts);
+    if (this.opts.useNewSimulator) {
+      log.debug(`New simulator is requested. If this is not wanted, set 'useNewSimulator' capability to false`);
+    } else {
+      // figure out the correct simulator to use, given the desired capabilities
+      let device = await getExistingSim(this.opts);
 
-    // check for an existing simulator
-    if (device) {
-      return {device, realDevice: false, udid: device.udid};
+      // check for an existing simulator
+      if (device) {
+        return {device, realDevice: false, udid: device.udid};
+      }
+
+      log.info('Simulator udid not provided');
     }
 
-    // no device of this type exists, so create one
-    log.info('Simulator udid not provided, using desired caps to create a new simulator');
+    // no device of this type exists, or they request new sim, so create one
+    log.info('Using desired caps to create a new simulator');
     if (!this.opts.platformVersion && this.iosSdkVersion) {
       log.info(`No platformVersion specified. Using latest version Xcode supports: '${this.iosSdkVersion}' ` +
                `This may cause problems if a simulator does not exist for this platform version.`);
       this.opts.platformVersion = this.iosSdkVersion;
     }
 
-    if (this.opts.noReset) {
```

Comment:
don't we need to consider noReset?

Reply:
I honestly could not figure out how this block ever got run. The original code was:
```js
let device = await getExistingSim(this.opts);

// check for an existing simulator
if (device) {
  return {device, realDevice: false, udid: device.udid};
}

// no device of this type exists, so create one
log.info('Simulator udid not provided, using desired caps to create a new simulator');
if (!this.opts.platformVersion && this.iosSdkVersion) {
  log.info(`No platformVersion specified. Using latest version Xcode supports: '${this.iosSdkVersion}' ` +
           `This may cause problems if a simulator does not exist for this platform version.`);
  this.opts.platformVersion = this.iosSdkVersion;
}

if (this.opts.noReset) {
  // Check for existing simulator just with correct capabilities
  let device = await getExistingSim(this.opts);
  if (device) {
    return {device, realDevice: false, udid: device.udid};
  }
}
```
Which means the `noReset` section is only different in that the `platformVersion`, if empty, has been set to the Xcode default.

Reply:
I'll put it back in, since it actually seems to have a use (https://github.com/appium/appium-xcuitest-driver/pull/520). Needs to be documented in the code, though, since it is unclear.

Reply:
Though I actually think moving the `platformVersion` check up a little is a better solution.